ledger/persistedaccts_list_test.go

```diff
@@ -0,0 +1,208 @@
+package ledger
+
+import (
+	"github.com/algorand/go-algorand/data/basics"
+	"testing"
+)
+
+func checkLen(list *persistedAccountDataList) int {
+	if isLenZero(list) {
+		return 0
+	}
+	return countListSize(&list.root)
+}
+
+func countListSize(head *persistedAccountDataListNode) (counter int) {
+	for i := head.next; i != head && i != nil; i = i.next {
+		counter += 1
+	}
+	return counter
+}
+
+func TestPersistedAccountDataList(t *testing.T) {
+	t.Run("single element list movements", testSingleElementListPositioning)
+
+	t.Run("multi-element list movements", testMultielementListPositioning)
+
+	t.Run("test remove", testRemove)
+
+	t.Run("test freelist usage", testFreeListMovement)
+}
+
+func testRemove(t *testing.T) {
+	t.Run("remove from list", removeFromListTest)
+}
+
+func removeFromListTest(t *testing.T) {
+	l := newPersistedAccountList()
+	e1 := l.pushFront(&persistedAccountData{addr: basics.Address{1}})
+	e2 := l.pushFront(&persistedAccountData{addr: basics.Address{2}})
+	e3 := l.pushFront(&persistedAccountData{addr: basics.Address{3}})
+	checkListPointers(t, l, []*persistedAccountDataListNode{e3, e2, e1})
+
+	l.remove(e2)
+	checkListPointers(t, l, []*persistedAccountDataListNode{e3, e1})
+	l.remove(e3)
+	checkListPointers(t, l, []*persistedAccountDataListNode{e1})
```

Comment:
could add a test for `move` method:
 - move to front/back at single element list
 - swap on two elements list